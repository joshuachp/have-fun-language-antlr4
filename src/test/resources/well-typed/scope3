fun f(f) {
   out(f);
   f = 10;
   out(f);
   f = 100;
   f = g(f);
   out(f);
   return g(f)
}

fun g(f) {
  g = 20;
  out(f);
  f = 200;
  out(g);
  out(f);
  return g
}

f = 99;
out(f(g(f)));
out(f)