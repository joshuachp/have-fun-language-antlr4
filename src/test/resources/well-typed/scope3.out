99
20
200
20
10
100
20
200
20
20
20
200
20
99